rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositoryImpl.scala

```diff
@@ -0,0 +1,218 @@
+package coop.rchain.rspace.nextgenrspace.history
+
+import cats.Applicative
+import cats.effect.Sync
+import cats.implicits._
+import coop.rchain.rspace.{
+  internal,
+  Blake2b256Hash,
+  DeleteContinuations,
+  DeleteData,
+  DeleteJoins,
+  HotStoreAction,
+  InsertContinuations,
+  InsertData,
+  InsertJoins,
+  Serialize,
+  StableHashProvider
+}
+import coop.rchain.rspace.internal._
+import scodec.{Attempt, Codec, DecodeResult, Decoder, Encoder, Err, SizeBound}
+import scodec.bits.BitVector
+import scodec.codecs.int32
+import HistoryRepositoryImpl.codecSeq
+
+final case class HistoryRepositoryImpl[F[_]: Sync, C, P, A, K](
+    history: History[F],
+    leafStore: ColdStore[F]
+)(implicit codecC: Codec[C], codecP: Codec[P], codecA: Codec[A], codecK: Codec[K])
+    extends HistoryRepository[F, C, P, A, K] {
+
+  def codecSeqC: Codec[Seq[C]] = codecSeq(codecC)
+
+  implicit def codecSeqSeqC: Codec[Seq[Seq[C]]] =
+    codecSeq(codecSeqC)
+
+  implicit def codecSeqDatum: Codec[Seq[internal.Datum[A]]] =
```

Comment:
I don't think this needs to be explicitly defined. Isn't it going to be derived implicitly?

Reply:
It was hard to convince the compiler these were the ones to pick. now that seq is aligned this might just work... will try.